src/main/kotlin/cacophony/semantic/syntaxtree/SyntaxTree.kt

```diff
@@ -125,29 +125,37 @@ sealed class Definition(
                 areEquivalentTypes(returnType, other.returnType)
     }
 
-    class ForeignFunctionDef(
+    class ForeignFunctionDeclaration(
         range: Pair<Location, Location>,
         identifier: String,
         type: Type.Functional,
         returnType: Type,
```

Comment:
Just a question, I see it's done this way in different places as well. Type.Functional already has returnType, so it seems like we have duplicate returnType here. Maybe one of them is not used, or something. It has too many occurrences in the code to check, so I'll just leave this comment and you may add this as an issue or maybe I don't understand something here.